File: nhglib-core/src/io/github/movementspeed/nhglib/runtime/fsm/interfaces/EngineConfigurationListener.java
package io.github.movementspeed.nhglib.runtime.fsm.interfaces;

import com.artemis.BaseSystem;
import com.badlogic.gdx.utils.Array;

/**
 * Created by Fausto Napoli on 02/11/2016.
 */
public interface EngineConfigurationListener {
    Array<BaseSystem> onConfigureEntitySystems();
}

File: nhglib-core/src/io/github/movementspeed/nhglib/core/fsm/interfaces/EngineConfigurationListener.kt
package io.github.movementspeed.nhglib.core.fsm.interfaces

import com.artemis.BaseSystem
import com.badlogic.gdx.utils.Array

/**
 * Created by Fausto Napoli on 02/11/2016.
 */
interface EngineConfigurationListener {
    fun onConfigureEntitySystems(): Array<BaseSystem>
}
